src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -391,6 +391,13 @@ private fun processQualifiedPathResolveVariants(
             if (processAll(exportedMacros(base), processor)) return true
         }
     }
+
+    // Proc macro crates are not allowed to export anything but procedural macros,
+    // and all possible macro exports are collected above.
+    if (base.containingCargoTarget?.isProcMacro == true) {
```

Comment:
Looks like we can move it into `if (base is RsMod)` block?

Reply:
Also looks like you should allow a case when `base.containingCargoTarget == path.containingCargoTarget`

Reply:
Do you mean that when we are resolving use items from crate A to crate A we should be able to see all functions?

Reply:
I've added extra check for it and additional test